discord/ext/commands/converter.py

```diff
@@ -159,6 +159,80 @@ class UserConverter(IDConverter):
 
         return result
 
+class MessageConverter(IDConverter):
+    """Converts to a :class:`discord.Message`.
+
+    The lookup strategu is as follows (in order):
+
+    1. Lookup by "{channel ID}-{message ID}" (got by shift-clicking on "Copy ID")
+    2. Lookup by message URI
+    3. Lookup by message ID (the message **must** be in the context channel)
+    """
+
+    async def convert(self, ctx, argument: str) -> discord.Message:
+        """|coro|
+
+        The method to override to do conversion logic.
+
+        If an error is found while converting, it is recommended to
+        raise a :exc:`.CommandError` derived exception as it will
+        properly propagate to the error handlers.
+
+        Parameters
+        -----------
+        ctx: :class:`.Context`
+            The invocation context that the argument is being used in.
+        argument: :class:`str`
+            The argument that is being converted.
+        """
+        if len(argument.split()) > 1:
+            raise BadArgument("The message must be an ID or an URI.")
+        exception = BadArgument('Message "{msg}" not found.'.format(msg=argument))
```

Comment:
I don't believe I use this idiom anywhere in the library, ergo nowhere do I make an exception object ahead of time and then raise it.